src/circuit/ecc/general_ecc/add.rs

```diff
@@ -0,0 +1,114 @@
+use super::AssignedPoint;
+use crate::circuit::integer::IntegerInstructions;
+use crate::circuit::main_gate::MainGateInstructions;
+use crate::circuit::{Assigned, AssignedCondition, AssignedInteger};
+use crate::circuit::ecc::general_ecc::{GeneralEccChip, GeneralEccInstruction};
+use halo2::arithmetic::{CurveAffine, FieldExt};
+use halo2::circuit::Region;
+use halo2::plonk::Error;
+
+impl<C: CurveAffine, F: FieldExt> GeneralEccChip<C, F> {
+    fn curvature(&self, region: &mut Region<'_, F>, a: &AssignedPoint<F>, offset: &mut usize) -> Result<(AssignedInteger<F>, AssignedCondition<F>), Error> {
+        let base_chip = self.base_field_chip();
+        // (3 * a.x^2 + self.a) / 2 * a.y
+        let xsqm = {
+            let xsq = base_chip.mul(region, &a.x, &a.x, offset)?;
+            let xsq2 = base_chip.add(region, &xsq, &xsq, offset)?;
+            base_chip.add(region, &xsq, &xsq2, offset)?
+            //base_chip.mul(region, &xsq, cst3, offset)?
+        };
+        let (curvature, icond) = {
+            let numerator = base_chip.add(region, &xsqm, &self.a, offset)?;
```

Comment:
Following the previous comment here we can enjoy `parameter_a` functiin combined with `add_constant` like below.

```
let numerator = base_chip.add_constant(region, &xsqm, &self.parameter_a(), offset)?;
```

Reply:
Done